clang/test/SemaSYCL/unique_stable_name.cpp

```diff
@@ -0,0 +1,174 @@
+// RUN: %clang_cc1 %s --std=c++17 -fsyntax-only -fsycl-is-device -Wno-unused -verify
+
+template <typename KernelName, typename KernelType>
+[[clang::sycl_kernel]] void kernel_single_task(KernelType kernelFunc) { // #1 #2 #3 #4 #5 #6 #7 #8 #9 #10 #11
+  kernelFunc();
+}
+
+template <typename Func>
+void kernel3func(const Func &F3) {
+  constexpr const char *F3_output = __builtin_unique_stable_name(F3);
+  // expected-error@#8{{kernel instantiation changes the result of an evaluated '__builtin_unique_stable_name'}}
+  // expected-note@+1{{in instantiation of function template specialization}}
+  kernel_single_task<class kernel3>(F3); // 8th instantiation throwing error
```

Comment:
I don't think the `Nth instantiation throwing error` comments add a whole lot -- perhaps they should be removed?